src/TagsInput/components/InputField.js

```diff
@@ -0,0 +1,134 @@
+// @flow
+
+import * as React from 'react';
+import { View, TextInput, Text } from 'react-native';
+import type { ViewLayoutEvent } from 'react-native/Libraries/Components/View/ViewPropTypes';
+import { StyleSheet } from '../../PlatformStyleSheet';
+
+type Props = {|
+  +onChangeText: (value: string) => void,
+  +value: string,
+  +onLayout: (e: ViewLayoutEvent) => void,
+  +maxWidth: number | string,
+  +minWidth: number,
+  +fontSize: number,
+  +disabled: ?boolean,
+  +placeholder: ?string,
+  +setRef: (el: any) => void, // @TODO find a way how to type this properly, the issue with mixing react and react-native
+|};
+
+type State = {|
+  componentWidth: number | string,
+|};
+
+class InputField extends React.PureComponent<Props, State> {
+  static defaultProps = {
+    maxWidth: 'auto',
+    minWidth: 'auto',
+  };
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      componentWidth: props.minWidth ?? 'auto',
+    };
+  }
+
+  handleOnLayout = (event: ViewLayoutEvent) => {
+    const { onLayout, maxWidth, minWidth, fontSize } = this.props;
+    const { width: inputWidth } = event.nativeEvent.layout;
+
+    const getWidth = width => {
+      const withBuffer = width + fontSize * 1.63;
```

Comment:
Could you explain where this 1.63 come from?

Reply:
This kind of magic, after experimenting with different value and this works best. I need this buffer because otherwise, it will be blink. Because it looks like the width of a character is a little bit different in TextInput against Text.

Reply:
Would it be possible to include that explanation as a comment in the code?

Reply:
comment added